apidef/oas/middleware.go

```diff
@@ -100,33 +116,36 @@ func (g *Global) Fill(api apidef.APIDefinition) {
 		g.CORS = nil
 	}
 
-	if g.PrePlugin == nil {
-		g.PrePlugin = &PrePlugin{}
-	}
-
-	g.PrePlugin.Fill(api)
-	if ShouldOmit(g.PrePlugin) {
+	if g.PrePlugins == nil {
+		g.PrePlugins = make(CustomPlugins, len(api.CustomMiddleware.Pre))
+		g.PrePlugins.Fill(api.CustomMiddleware.Pre)
 		g.PrePlugin = nil
 	}
 
-	if g.PostAuthenticationPlugin == nil {
-		g.PostAuthenticationPlugin = &PostAuthenticationPlugin{}
+	if ShouldOmit(g.PrePlugins) {
+		g.PrePlugins = nil
 	}
 
-	g.PostAuthenticationPlugin.Fill(api)
-	if ShouldOmit(g.PostAuthenticationPlugin) {
+	if g.PostAuthenticationPlugins == nil {
+		g.PostAuthenticationPlugins = make(CustomPlugins, len(api.CustomMiddleware.PostKeyAuth))
```

Comment:
The allocation should happen inside Fill (much like other Fill/ExtractTo funcs)

Reply:
`CustomPlugins` is an alias to `[]CustomPlugin`, so the allocations should happen before fill